### DISCUSSION TITLE
Unnecessary system tests for filters

### DISCUSSION THREAD

**Discussion by themudassarhassan** [joemasilotti/railsdevs.com#338] [General]
Hey @joemasilotti 
I was working on the test coverage of filtering developers by their bio or hero feature and noticed the system tests in `test/system/filters_test.rb` does not provide any value/confidence for writing tests because what these test cases are testing for is already covered in the integration tests in `test/integration/developers_test.rb`.

Let me explain why test cases for filters are not providing any value, these tests are asserting that when the user clicks on some filter that filter's value should be added in the query parameter of the `URL` on the other hand in integration tests of `developers_test.rb` we are visiting the `URL` with the required query parameters and then asserting on the returned results so if something with the query parameter of the `URL` goes side ways the integration tests will fail (deprecating the need for having separate tests for filters)

Also as the order of query parameters does not matter in the `URL` and filter test cases are comparing strings for `URL` matching so if we don't keep the order the assertion will fail (false negative test). For example:
<img width="1440" alt="Screenshot 2022-03-19 at 8 20 17 PM" src="https://user-images.githubusercontent.com/31462903/159127034-3f17b045-fe35-4010-8075-e8bd8d0192a4.png">

Please let me know what do you think about it or if I make sense altogether :D

**Comment by joemasilotti** [comment]
I see your point! Feel free to open a PR that tests the filters in a better way.

**Reply by themudassarhassan** [reply]
I think test cases in `test/system/filters_test.rb` are not needed for all reasons mentioned above.

**Reply by joemasilotti** [reply]
Do you propose an alternative test or just removing them?

**Reply by themudassarhassan** [reply]
Okay so first I thought we should remove these system tests but after investigating again I came to know these system tests are covering the user's interaction with the application (like selecting a checkbox, clicking button) so we cannot remove these but we can improve the assertion path to not compare strings (because the order of query params does not matter). I'm opening a PR to address this.

**Reply by joemasilotti** [reply]
PR #340 merged. Thanks for bringing this up!
